proto/process/agent.proto

```diff
@@ -720,6 +720,21 @@ message Connection {
 	uint32 stateIndex = 46;
 }
 
+message FailedConnection {
+	int32 pid = 1;
+	Addr laddr = 2; // Local address
+	Addr raddr = 3; // Remote address
```

Comment:
Since this obect is always scoped under a connection, this seems duplicative (and bad on on memory use)

Reply:
It is not scoped under a `Connection` but a `FailedConnection`. This choice was made because of Go allocating the memory necessary for all the fields in a struct. Using the `Connection` type for failed connection would use a lot of memory while using only a few fields.

Reply:
we could address this issue in another PR as probably gogoprotobuf nullable (TBC) could help us here  (for backend only fields and failedConnections specific) https://pkg.go.dev/github.com/gogo/protobuf/gogoproto?utm_source=godoc#hdr-More_Canonical_Go_Structures

Reply:
Could we just index into the `Connections` field instead? `FailedConnection` could just be an index + failure count.

Reply:
Looking at the other PR for the agent, I think this should be ok since the `Connections` and `FailedConnections` fields should have mutually exclusive contents.